tools/cleanup_mri_tables_for_19-0_release.php

```diff
@@ -0,0 +1,358 @@
+        backupEntries($select_ID_query, $table_name, $FK_field);
+    }
+
+
+    // clean up the table $table_name based on answer provided by the user
+    cleanTable(
+        $table_name,
+        $FK_field,
+        $select_ID_query,
+        $delete_answer
+    );
+    
+}
+
+/**
+ * selects orphan entries base on select query
+ *
+ * @param string $query query to use to select orphans
+ *
+ * @return array result of the select query
+ *
+ */
+function selectOrphan($query)
+{
+    $db = Database::singleton();
+
+    $result = $db->pselect($query, []);
+
+    return $result;
+}
+
+/**
+ * prints the list of found orphan entries in the table
+ *
+ * @param string $table_name  name of the table
+ * @param array  $orphan_list array with the list of orphan entries found]
+ *
+ * @return nothing
+ */
+function printOrphan($table_name, $orphan_list)
+{
+    print "\n\n\nList of orphan entries for " . $table_name . "\n";
+    print "__________________________________________________\n";
+    print_r($orphan_list);
+}
+
+/**
+ * ask whether the user wants to delete orphans from the table
+ *
+ * @param string $table_name name of the table in which to delete orphans
+ * @param string $FK_field   foreign key to set to null if don't want to delete
+ * orphans
+ *
+ * @return string  answer from the user to the question
+ */
+function askToDelete($table_name, $FK_field)
+{
+    // create question to be printed in the terminal
+    $message = "\nDo you want to delete the orphan entries from $table_name?
+      'Y' to delete from database and keep a copy in a .sql file 
+      'N' to keep the entry but set $FK_field to NULL \n";
+    print $message;
+
+    // grep answer from stdin and remove carriage return from it
+    $handle = fopen("php://stdin", "r");
+    $answer = fgets($handle);
+    $answer = str_replace("\n", "", $answer);
+    $answer = str_replace("\r", "", $answer);
+
+    // return user's answer
+    return $answer;
+}
+
+/**
+ * backup the orphan entries into a text file
+ *
+ * @param string $selectID   select IDs query
+ * @param string $table_name name of the table to backup
+ * @param string $FK_field   name of the ID field to use backup
+ *
+ * @return nothing
+ */
+function backupEntries($selectID, $table_name, $FK_field)
+{
+    // grep the IDs to backup from the database based on query stored in
+    // $selectID. This will be given as an argument to mysqldump using
+    // --where option
+    $IDs = generateIdList($selectID, $FK_field);
+    $where = $FK_field . " IN (" . $IDs . ")";
+
+    // grep database connection information from NDB_Config for mysqldump
+    $config = NDB_Config::singleton();
+    $config->load();
+    $database = $config->getSettingFromXML("database");
+
+    // create the mysqldump query to back the orphan entries to be deleted
+    $sqldump = "mysqldump " .
+        "-u " . $database['username'] . " " .
+        "-h " . $database['host']     . " " .
+        "-p"  . $database['password'] . " " .
```

Comment:
These need to be shell escaped. See https://secure.php.net/manual/en/function.escapeshellarg.php